src/python/pants/jvm/target_types.py

```diff
@@ -267,23 +267,80 @@ class JvmProvidesTypesField(StringSequenceField):
     )
 
 
-class JvmArtifactExcludeDependenciesField(StringSequenceField):
-    alias = "excludes"
-    help = help_text(
-        """
-        A list of unversioned coordinates (i.e. `group:artifact`) that should be excluded
+@dataclass(frozen=True)
+class JvmArtifactExclusionRule:
```

Comment:
`Rule` is an overloaded term here: it's _probably_ not going to be easily confused with `@rule`, but it seems like this could also just be called `JvmArtifactExclusion`.